### PR TITLE
repo-updater: Use code host URL as key for rate limit registry

### PR DESCRIPTION
**Pull request by ryanslade**
Related to: https://github.com/orgs/sourcegraph/teams/core-services/discussions/4

I'll do a follow up PR that adds the warning when more than one external service for the same host has been configured with a rate limiter.

### CODE REVIEW

File: cmd/repo-updater/repos/types.go [hunk 1]
```diff
@@ -910,117 +946,145 @@ func (es ExternalServices) With(opts ...func(*ExternalService)) ExternalServices
...
-		CreatedAt:   apiService.CreatedAt,
-		UpdatedAt:   apiService.UpdatedAt,
-		DeletedAt:   time.Time{},
-	}
-	if apiService.DeletedAt != nil {
-		svc.DeletedAt = *apiService.DeletedAt
-	}
-	return r.updateRateLimiter(svc)
+func (r *RateLimiterRegistry) HandleExternalServiceSync() error {
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
```
Comment (keegancsmith) [inline cmd/repo-updater/repos/types.go:989]: why the need to protect it with a timeout? From reading the function it doesn't seem to contact external code hosts, or am I wrong about that?

Comment (ryanslade) [reply cmd/repo-updater/repos/types.go:989]: That's correct, it only communicates with our database so I assume it's safe to remove the timeout

File: cmd/repo-updater/repos/types.go [hunk 2]
```diff
@@ -910,117 +946,137 @@ func (es ExternalServices) With(opts ...func(*ExternalService)) ExternalServices
 	return clone
 }
 
+type ExternalServicesLister interface {
```
Comment (tsenart) [inline cmd/repo-updater/repos/types.go:949]: Does this interface need to be exported?

Comment (tsenart) [reply cmd/repo-updater/repos/types.go:949]: If so, please add a Go doc.

File: cmd/repo-updater/repos/types.go [hunk 3]
```diff
@@ -910,117 +946,137 @@ func (es ExternalServices) With(opts ...func(*ExternalService)) ExternalServices
...
+	l := r.rateLimiters[baseURL]
+	if l == nil {
+		l = rate.NewLimiter(rate.Inf, 100)
+		r.rateLimiters[baseURL] = l
 	}
+	return l
+}
 
-	svcs, err := store.ListExternalServices(ctx, StoreListExternalServicesArgs{})
+// SyncRateLimiters will sync all rate limiters with current config.
```
Comment (tsenart) [inline cmd/repo-updater/repos/types.go:986]:
```suggestion
// SyncRateLimiters syncs all rate limiters with current config.
```

**Pull request merged**